cypress/e2e/dashboard.cy.ts

```diff
@@ -1,16 +1,24 @@
-import { Budget } from '../../src/types'
-import { createBudget, createEnvelope } from '../support/setup'
+import { Budget, Envelope } from '../../src/types'
+import {
+  createBudget,
+  createAccount,
+  createEnvelope,
+  createTransaction,
+} from '../support/setup'
 
 describe('Dashboard', () => {
   beforeEach(() => {
     // prepare & select a budget
     cy.wrap(createBudget({ name: 'Dashboard Test' })).then((budget: Budget) => {
+      cy.wrap(budget).as('budget')
       cy.wrap(
         Cypress.Promise.all([
           createEnvelope({ name: 'First Envelope' }, budget),
           createEnvelope({ name: 'Second Envelope' }, budget),
         ])
-      ).then(() => {
+      ).then(([firstEnvelope, secondEnvelope]: Array<Envelope>) => {
+        cy.wrap(firstEnvelope).as('firstEnvelope')
+        cy.wrap(secondEnvelope).as('secondEnvelope')
```

Comment:
Do we really need the envelopes in the beforeEach?

Reply:
Nope, moving them to the specific test